code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -30,31 +30,89 @@ var/global/list/parasites = list() //all currently existing/living guardians
-	melee_damage_lower = 15
-	melee_damage_upper = 15
+	obj_damage = 0
+	melee_damage_lower = 0
+	melee_damage_upper = 0 // all 0 by default until abilities come into play
 	butcher_results = list(/obj/item/weapon/ectoplasm = 1)
 	AIStatus = AI_OFF
 	dextrous_hud_type = /datum/hud/dextrous/guardian //if we're set to dextrous, account for it.
+	ranged_cooldown_time = 0 //changed in the abilities datum
 	var/list/guardian_overlays[GUARDIAN_TOTAL_LAYERS]
 	var/reset = 0 //if the summoner has reset the guardian already
 	var/cooldown = 0
 	var/mob/living/summoner
-	var/range = 10 //how far from the user the spirit can be
+
+	var/obj/item/internal_storage //what we're storing within ourself
+
+	var/list/abilities = list()
+	var/list/current_abilities = list()
+
+	var/range = 0 //how far from the user the spirit can be
+	var/has_mode = FALSE
 	var/toggle_button_type = /obj/screen/guardian/ToggleMode/Inactive //what sort of toggle button the hud uses
 	var/datum/guardianname/namedatum = new/datum/guardianname()
 	var/playstyle_string = "<span class='holoparasite'>You are a standard Guardian. You shouldn't exist!</span>"
 	var/magic_fluff_string = "<span class='holoparasite'>You draw the Coder, symbolizing bugs and errors. This shouldn't happen! Submit a bug report!</span>"
 	var/tech_fluff_string = "<span class='holoparasite'>BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!</span>"
 	var/carp_fluff_string = "<span class='holoparasite'>CARP CARP CARP SOME SORT OF HORRIFIC BUG BLAME THE CODERS CARP CARP CARP</span>"
 
+/mob/living/simple_animal/hostile/guardian/verb/Battlecry()
+	set name = "Set Battlecry"
+	set category = "Guardian"
+	set desc = "Choose what you shout as you attack people."
+	var/input = stripped_input(src,"What do you want your battlecry to be? Max length of 6 characters.", ,"", 7)
+	if(input)
+		for(var/datum/guardian_abilities/I in current_abilities)
+			I.battlecry = input
+
 /mob/living/simple_animal/hostile/guardian/New(loc, theme)
-	parasites |= src
 	setthemename(theme)
+	for(var/type in abilities)
+		var/datum/guardian_abilities/G = new type
+		G.user = summoner
+		G.stando = src
+		G.handle_stats()
+		current_abilities += G
+	parasites |= src
+	..()
+
+/mob/living/simple_animal/hostile/guardian/Shoot()
+	. = ..()
+	for(var/datum/guardian_abilities/I in current_abilities)
+		I.ranged_attack()
+
+/mob/living/simple_animal/hostile/guardian/AttackingTarget()
+	if(src.loc == summoner)
+		src << "<span class='danger'><B>You must be manifested to attack!</span></B>"
+		return 0
+	else
+		for(var/datum/guardian_abilities/I in current_abilities)
+			I.ability_act()
+		..()
+		return 1
+
+/mob/living/simple_animal/hostile/guardian/AltClickOn(atom/movable/A)
+	for(var/datum/guardian_abilities/I in current_abilities)
+		I.alt_ability_act(A)
+
+/mob/living/simple_animal/hostile/guardian/Crossed(atom/movable/A)
+	..()
+	for(var/datum/guardian_abilities/I in current_abilities)
+		I.bump_reaction(A)
```

Comment:
no it's supposed to call in all 3 that's how the code was originally designed it's just named bump_reaction for the simplicity of the term